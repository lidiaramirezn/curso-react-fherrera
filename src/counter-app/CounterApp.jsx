import PropTypes from 'prop-types';
import './counter.css';
import { useState } from 'react';

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => {
    setCounter( counter + 1 );
    // setCounter( (c) => c + 1 )
  }

  return (
    <>
      <h1>Contador</h1>
      <p>{ counter }</p>

      <button onClick= { handleAdd }>
        +1
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}
