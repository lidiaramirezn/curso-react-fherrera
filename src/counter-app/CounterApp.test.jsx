/**
 * @jest-environment jsdom
 */
 import { render, screen, fireEvent } from '@testing-library/react';
import { CounterApp } from './CounterApp';

describe('Pruebas en el <CounterApp />', () => {
  const initialValue = 10;

  test('Dee de hacer match con el snapshot ', () => {
    const { container } = render(<CounterApp value={initialValue} />)
    expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar el valor inicial de 100 <CounterApp value={100}>', () => {
        
    render( <CounterApp value={ 100 } /> );
    expect( screen.getByText(100) ).toBeTruthy();

  });

  test('debe de funcionar el botÃ³n de reset', () => {
        
    render( <CounterApp value={ 355 } /> );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    expect( screen.getByText( 355 ) ).toBeTruthy();

  });
});