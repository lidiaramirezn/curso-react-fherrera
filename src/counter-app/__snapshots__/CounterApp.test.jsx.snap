// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Dee de hacer match con el snapshot  1`] = `
<div>
  <h1>
    Contador
  </h1>
  <p>
    10
  </p>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
